src/solver/variable/state.h

```diff
@@ -60,6 +61,10 @@ class ThermalState
 
         //! Minimum power of all clusters for the current hour in the year
         std::vector<double> PMinOfClusters;
+
+        unsigned int unitCountLastHour = 0;
+        double productionLastHour = 0;
```

Comment:
I think these should be vectors. One coefficient for each thermal cluster within the area.

Reply:
Otherwise you'll only take the last cluster into account, with overwriting, etc.